### PR TITLE
feat: Add auto-injection support for Layotto container at namespace s…

### PR DESCRIPTION
**Pull request by xiaoxiang10086**
creating clusterrole and clusterolebinding to allow Kubernetes user list/get namespaces

### CODE REVIEW

File: yaml/namespace/layotto-namespace.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: layotto-namespace
+  labels:
+    mosn.io/layotto-inject: "true"
```
Comment (Xunzhuo) [inline yaml/namespace/layotto-namespace.yaml:6]: A new line at the end of yaml is always a better style

File: yaml/namespace/layotto-pod-inject.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+kind: Pod
+metadata:
+  name: layotto-pod-inject
+  namespace: layotto-namespace
+spec:
+  containers:
+    - name: nginx
+      image: nginx
+      ports:
+        - containerPort: 80
```
Comment (Xunzhuo) [inline yaml/namespace/layotto-pod-inject.yaml:11]: ditto

File: yaml/namespace/layotto-pod.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  namespace: layotto-namespace
+  labels:
+    # Add the following label to disable Layotto Sidecar injection for this pod
+    mosn.io/sidecar-inject: "false"
+spec:
+  containers:
+    - name: nginx
+      image: nginx
+      ports:
+        - containerPort: 80
```
Comment (Xunzhuo) [inline yaml/namespace/layotto-pod.yaml:14]: ditto

File: yaml/namespace/layotto-role-binding.yaml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+metadata:
+  name: global-namespaces-viewer-binding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: global-namespaces-viewer
+subjects:
+  - kind: ServiceAccount
+    name: default
+    namespace: default
```
Comment (Xunzhuo) [inline yaml/namespace/layotto-role-binding.yaml:21]: ditto

**Review by Xunzhuo — COMMENTED** [review]
Some nits, generally LGTM.

But please add an example of deployment not just single pod.

File: yaml/namespace/layotto-role-binding.yaml (hunk 4 above)
Comment (xiaoxiang10086) [reply yaml/namespace/layotto-role-binding.yaml:21]: Ok, I will fix the above issue in a subsequent commit

**Comment by xiaoxiang10086** [comment]
It seems that adding a new line does not work, Github will delete the extra empty line

File: yaml/namespace/layotto-inject-deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,22 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: layotto-inject-deployment
+  namespace: layotto-namespace
+  labels:
+    app: layotto
+spec:
+  replicas: 1
```
Comment (Xunzhuo) [inline yaml/namespace/layotto-inject-deployment.yaml:9]: can we set multiple replicas here not just 1?

**Comment by Xunzhuo** [comment]
> It seems that adding a new line does not work, Github will delete the extra empty line

It is already fixed.

File: yaml/namespace/layotto-inject-deployment.yaml (hunk 5 above)
Comment (xiaoxiang10086) [reply yaml/namespace/layotto-inject-deployment.yaml:9]: OK,  I will modify it to 3

**Review by Xunzhuo — APPROVED** [review]
/lgtm

**Pull request merged**